planet/src/main/webapp/js/authentication.js

```diff
@@ -67,27 +66,52 @@ function resetForm(elementsClass) {
     }
 }
 
-function signUp() {
+async function generateUsername(displayName) {
+    const names = displayName.split(' ');
+    const existingUsernames = [];
+    let username = names[0].toLowerCase() + '.' + names[1].toLowerCase();
```

Comment:
Just curious: do usernames usually contain dot? if people can access profiles through the username, I feel like https://example.com/profile/johnsmith is more common than https://example.com/profile/john.smith?
LGTM otherwise

Reply:
Some of them do some of them don't. LinkedIn puts dashes and Facebook uses dots!